### PR TITLE
[WAIT-FRONT] Add "status_max_length" instance parameter

### PR DESCRIPTION
**Pull request by alkemist**
Add the ability to edit the maximum number of characters per status

### CODE REVIEW

**Comment by pixocode** [comment]
#1781

File: app/views/admin/settings/index.html.haml [hunk 1]
```diff
@@ -38,3 +38,7 @@
         %strong= t('admin.settings.registrations.closed_message.title')
         %p= t('admin.settings.registrations.closed_message.desc_html')
       %td= best_in_place @settings['closed_registrations_message'], :value, as: :textarea, url: admin_setting_path(@settings['closed_registrations_message'])
+    %tr
+      %td
+        %strong Status max length
```
Comment (ineffyble) [inline app/views/admin/settings/index.html.haml]: Can you change this to be internationalised?

**Review by ineffyble — CHANGES_REQUESTED** [review]

File: app/views/admin/settings/index.html.haml (hunk 1 above)
Comment (pixocode) [inline app/views/admin/settings/index.html.haml]: Fixed :)

**Comment by pixocode** [comment]
If someone is skilled reactjs, help would be greatly appreciated ^^"

**Review by ineffyble — APPROVED** [review]

**Comment by pixocode** [comment]
It lacks a "front" part (which I did not succeed to do) so that the parameter is repercussion on the toot form :/

**Comment by rixx** [comment]
So the only part missing is to replace the `500` [here](https://github.com/tootsuite/mastodon/blob/master/app/assets/javascripts/components/features/compose/components/compose_form.jsx#L176)?

**Comment by pixocode** [comment]
@rixx yes, but I don't know how to retrieve the parameter from the back (with reactjs :/)

**Comment by BrunoSpy** [comment]
you can look at 
https://gist.github.com/BrunoSpy/6581f106a15f5d830f3e2289002da2f5
We need a way to get the value of Setting.status_max_length.to_i in the two jsx files mentioned in the previous gist. But I don't know how.

**Comment by BrunoSpy** [comment]
And every locale file must be changed too ("500" is hardcoded there too).

**Review by nightpool — CHANGES_REQUESTED** [review]
needs the above concerns addressed, at least. (front-end changes, locale changes)

but I'm also not sure this is a good feature for the project to have, from a meta perspective. it's pretty easy for admins to change this themselves, to be sure, but I don't know that making this particular setting *easier* to change is a good idea in the long run.

**Comment by rixx** [comment]
Could you expand upon this, @nightpool? Particularly on the "why"?

**Comment by Gargron** [comment]
Closing due to inactivity.

**Pull request closed without merging**